backend/mobilityAI/Controllers/MobilityAIController.cs

```diff
@@ -129,6 +135,92 @@ public JsonResult GetRangeGyroscope(long start, long end, int patientId)
             return new JsonResult(dataRange.ToList());
         }
 
+        /// <summary>
+        /// Add a single file with accelerometer and gyroscope data into the database and into machine learning server
+        /// TODO: Handle larger files (ie: don't wait for the files to be added to the database before returning 200 or else the client will timeout)
+        /// </summary>
+        /// <param name="DataFile">
+        /// The file input for the data to be analyzed
+        /// </param>
+        /// <param name="patientID">
+        /// The patient id from which this data came from
+        /// </param>
+        [HttpPost("AddDataSingle", Name = "AddDataSingle")]
+        public async Task<IActionResult> AddDataSingle(IFormFile DataFile, int patientId) 
+        {
+            List<String> accel = new List<string>();
+            List<String> gyro = new List<string>();
+            string ac = "epoch (ms),time (-13:00),elapsed (s),x-axis (g),y-axis (g),z-axis (g)" + Environment.NewLine;
+            string gy = "epoch (ms),time (-13:00),elapsed (s),x-axis (deg/s),y-axis (deg/s),z-axis (deg/s)" + Environment.NewLine;
+
+            using (var reader = new StreamReader(DataFile.OpenReadStream()))
+            {
+                while (reader.Peek() >= 0)
+                {
+                    String res = reader.ReadLine();
+                    if(String.Equals(res.Substring(0,2), "a,")) { accel.Add(res.Substring(2)); ac += res.Substring(2)+ Environment.NewLine; }
+                    if(String.Equals(res.Substring(0,2), "g,"))  { gyro.Add(res.Substring(2)); gy += res.Substring(2)+ Environment.NewLine; }
+                }
+            }
+
+            var AccelerometerObjects = accel.Select(line => line.Split(","))
```

Comment:
Can we merge this part with `AddData`? Lots of repetition.

Reply:
I was planning on merging this when we reworked the API, since we'd need to redo the testing for it anyways.